scipy/stats/_continuous_distns.py

```diff
@@ -9155,7 +9154,6 @@ def _log_diff(log_p, log_q):
 
 def _log_gauss_mass(a, b):
     """Log of Gaussian probability mass within an interval"""
-    a, b = np.atleast_1d(a), np.atleast_1d(b)
     a, b = np.broadcast_arrays(a, b)
```

Comment:
AFAICS this will bread indexing below if a and b are scalars

Reply:
But it doesn't. 
```python3
import numpy as np
a, _= np.broadcast_arrays(1, 2)
a[True]  # array([1])
```

This probably doesn't fix the deprecation issue, though. The problem is probably that the function returns a 1D array when the input is a scalar. The distribution infrastructure _sometimes_ converts scalar arguments into 1D arrays before passing them into the private functions, and makes sure that it returns scalars if need be, but it's inconsistent about this, so we might have to be more carefuly about the output shape within the function.

Reply:
ok, interesting and weird
we cannot index into a scalar but we can apply a mask.

```
a, b = np.broadcast_arrays(1, 2)
a.shape, b.shape
((), ())
 = 5
a[0] = 5
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
~\AppData\Local\Temp\ipykernel_30048\543285221.py in <module>
----> 1 a[0] = 5

IndexError: too many indices for array: array is 0-dimensional, but 1 were indexed

np.__version__
'1.21.5'

a[a>0] = 1
a
array(1)
​```

Is this reliable behavior for the mask (boolean indexing), or do we get another round of numpy deprecations (if I switch to using this)?
(I just wrote a PR for statsmodels a few days ago adding many atleast_1d)

Reply:
I wouldn't switch to this across the board. But I think it's fine for here.